pallets/gas/src/lib.rs

```diff
@@ -55,6 +56,7 @@ impl Default for ValueType {
 pub struct ValueNode {
     pub spec_refs: u32,
     pub unspec_refs: u32,
+    pub reserved_refs: u32,
```

Comment:
> what's the need of having refs for ReservedLocal nodes when they aren't used to return any value to upstream node (like SpecifiedLocal does), nor must be used to block parent node consumption?

yeah I'm thinking of this at the current point, if the value we dispatch to this node is just the multiple of MessageThreshold, there should be no reminders after consuming.

Reply:
I'm not sure, but semantics of `ReservedLocal`, which is a gas node type for mailbox messages, seems to be the following: 
1. **It can be created** by calling `cut` on any node of the gas tree.
2. **No splits** on it are possible, because mailbox message isn't executed in the context of the program, so it can generate new messages, therefore, new gas nodes. When message is in mailbox - it's done, can be only read and it's value can be claimed.
3. **No spends** on it are possible for reason state earlier - you can't partially read message from mailbox or partially claim value from it, while spend in most cases is about it (partial consumption of the gas in the node).
4. **It can be consumed.** It's consumption mustn't block consumption of other nodes in the tree for reasons stated earlier about spending value and splits.

Therefore, solution can be even simpler:
- just store the node in the storage without being a structural part of the tree;
- such node must know the `origin`'s address to return value back to him;
In storage it can be something like `GasTreeStorage { key: MessageId, value: ReservedLocalNode(origin_address, amount) }`.

Even more radical solution suggested by @ekovalev can be to just have a simple `StorageValue` type storage which stores gas values of mailbox messages.

Reply:
Regardless of the motivation for this particular change, I'm still generally against complicating the gas tree.

The original purpose of it was to represent the **computational** budget a smart contract can have at its disposal to run its code. In my option all other use cases (including rent payment for keeping a message in mailbox) should be handled differently.

**Note**: this is different from paying rent while staying in Wait List because waitlisted messages represent futures in `pending` state, that is the life cycle of such message is not yet over at that point. As opposed to messages in mailbox - those ones are `ready` and can neither spawn new computations nor should be blocking release of reserved funds to the original sender. But it's a conceptual discussion - has nothing to do with current implementation.

Reply:
> 3. No spends on it are possible for reason stated earlier - you can't partially read message from mailbox or partially claim value from it, while spend in most cases is about it (partial consumption of the gas in the node).

but we may need to burn the value of it on each thershold?

> The original purpose of it was to represent the computational budget a smart contract can have at its disposal to run its code. In my option all other use cases (including rent payment for keeping a message in mailbox) should be handled differently.

make sense

Reply:
> > 3. No spends on it are possible for reason stated earlier - you can't partially read message from mailbox or partially claim value from it, while spend in most cases is about it (partial consumption of the gas in the node).
> 
> but we may need to burn the value of it on each thershold?

Oh that's right. So `spend` is possible in case we implement current approach to charging a rent for being placed in mailbox. But basing on https://github.com/gear-tech/gear/pull/1095/files#r902406505 and https://github.com/gear-tech/gear/pull/1095/files#r902445764 I believe we can implement it differently